.github/workflows/main.yml

```diff
@@ -189,6 +189,79 @@ jobs:
         TEST_CASE: ${{ matrix.TEST_CASE }}
         TARGET: ${{ matrix.TARGET }}
 
+  opentitan-earl-grey-sim:
+    runs-on: ubuntu-latest
+    needs: build-binaries
+    env:
+      CC: gcc-9
+      CXX: g++-9
+
+    steps:
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get update -qq
+        sudo apt install -y g++-9 build-essential cmake tclsh ant default-jre swig google-perftools libgoogle-perftools-dev python3 python3-dev uuid uuid-dev tcl-dev flex libfl-dev libelf-dev ninja-build
+
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 1
+
+    - uses: actions/setup-python@v2
+      with:
+         python-version: '3.7'
+
+    - name: Download binaries
+      uses: actions/download-artifact@v2
+      with:
+        name: binaries
+
+    # See https://github.com/actions/upload-artifact/issues/38
+    - name: Extract
+      run: tar -xf binaries.tar
+
+    - name: Prepare output directory
+      run: mkdir output_uhdm
+
+    - name: Build & Test (vanilla verilator)
+      run: |
+        pip install virtualenv
+        ./uhdm-integration/.github/ci.sh
+        ./uhdm-integration/build/lowrisc_systems_top_earlgrey_verilator_0.1/sim-verilator/Vtop_earlgrey_verilator --meminit=rom,uhdm-tests/opentitan/boot_rom_sim_verilator.elf --meminit=flash,uhdm-tests/opentitan/hello_world_sim_verilator.elf | tee output_uhdm/vanilla_verilator_simulation.log
+        cp uart0.log output_uhdm/vanilla_verilator_uart.log
+        cp trace_core_00000000.log output_uhdm/vanilla_verilator_trace.log
+      env:
+        TARGET: verilator/earl-grey
+        TEST_CASE: tests/opentitan
+
+    - name: Build & Test (uhdm verilator)
+      run: |
+        pip install virtualenv
+        ./uhdm-integration/.github/ci.sh
+        ./uhdm-integration/build/lowrisc_systems_top_earlgrey_verilator_0.1/sim-verilator/Vtop_earlgrey_verilator --meminit=rom,uhdm-tests/opentitan/boot_rom_sim_verilator.elf --meminit=flash,uhdm-tests/opentitan/hello_world_sim_verilator.elf | tee output_uhdm/uhdm_verilator_simulation.log
+        cp uart0.log output_uhdm/uhdm_verilator_uart.log
+        cp trace_core_00000000.log output_uhdm/uhdm_verilator_trace.log
+      env:
+        TARGET: uhdm/verilator/earl-grey
+        TEST_CASE: tests/opentitan
+
+    - name: Compare results
+      run: |
+        diff output_uhdm/vanilla_verilator_uart.log output_uhdm/uhdm_verilator_uart.log > output_uhdm/uart.diff
```

Comment:
Are we expecting, that pipeline will fail, if there is diff found between outputs (and I think we should fail pipeline on any diff in outputs)? If yes, I think we shouldn't redirect output to the file as it won't get uploaded as an artifact and the output won't be printed on the console.

Reply:
Removed redirecting  output to files